atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifierTest.java

```diff
@@ -56,209 +60,250 @@ public void beforeEach() {
                 .isInstanceOf(IllegalStateException.class);
     }
 
     @Test
     public void returnSameKsDefIfNodeTopologyChecksIgnored() throws TException {
-        setTopology(createDetails(DC_1, RACK_1, HOST_1));
-        when(config.replicationFactor()).thenReturn(7);
-        when(config.ignoreNodeTopologyChecks()).thenReturn(true);
+        CassandraServersConfig cassandraServersConfig =
+                setTopologyAndGetServersConfig(createDetails(DC_1, RACK_1, HOST_1));
+        CassandraKeyspaceVerifierConfig cassandraKeyspaceVerifierConfig =
+                getCassandraKeyspaceVerifierConfigBuilderWithDefaults()
+                        .servers(cassandraServersConfig)
+                        .replicationFactor(7)
+                        .ignoreNodeTopologyChecks(true)
+                        .build();
+
         KsDef ksDef = new KsDef("test", CassandraConstants.SIMPLE_STRATEGY, ImmutableList.of());
         ImmutableMap<String, String> strategyOptions =
                 ImmutableMap.of(CassandraConstants.REPLICATION_FACTOR_OPTION, "1", DC_1, "7");
         ksDef.setStrategy_options(strategyOptions);
 
-        ksDef = CassandraVerifier.checkAndSetReplicationFactor(client, ksDef, config);
+        ksDef = CassandraVerifier.checkAndSetReplicationFactor(client, ksDef, cassandraKeyspaceVerifierConfig);
         assertThat(ksDef.strategy_class).isEqualTo(CassandraConstants.SIMPLE_STRATEGY);
         assertThat(ksDef.strategy_options).containsExactlyInAnyOrderEntriesOf(strategyOptions);
     }
 
     @Test
     public void networkStrategyMultipleDcsSucceeds() throws TException {
-        setTopology(
+        CassandraServersConfig cassandraServersConfig = setTopologyAndGetServersConfig(
                 createDetails(DC_1, RACK_1, HOST_1),
                 createDetails(DC_1, RACK_1, HOST_2),
                 createDetails(DC_1, RACK_1, HOST_3),
                 createDetails(DC_2, RACK_1, HOST_4));
-        when(config.replicationFactor()).thenReturn(3);
+        CassandraKeyspaceVerifierConfig cassandraKeyspaceVerifierConfig =
+                getCassandraKeyspaceVerifierConfigBuilderWithDefaults()
+                        .servers(cassandraServersConfig)
+                        .build();
+
         KsDef ksDef = new KsDef("test", CassandraConstants.NETWORK_STRATEGY, ImmutableList.of());
         ImmutableMap<String, String> strategyOptions =
                 ImmutableMap.of(CassandraConstants.REPLICATION_FACTOR_OPTION, "3", DC_1, "3", DC_2, "3");
         ksDef.setStrategy_options(strategyOptions);
 
-        ksDef = CassandraVerifier.checkAndSetReplicationFactor(client, ksDef, config);
+        ksDef = CassandraVerifier.checkAndSetReplicationFactor(client, ksDef, cassandraKeyspaceVerifierConfig);
         assertThat(ksDef.strategy_options).containsExactlyInAnyOrderEntriesOf(strategyOptions);
     }
 
     @Test
     public void differentRfThanConfigThrows() {
+        boolean ignoreDatacentreConfigurationChecks = false;
```

Comment:
is this different to the previous behaviour?

Reply:
No, but a good flag. The test previously was depending on mockito implicitly mocking a primitive boolean as false by default, even though the mock on `ignoreDatacentreConfigurationChecks` wasn't declared. 

I've just made that dependency clear.